cognite/client/credentials.py

```diff
@@ -23,6 +23,39 @@ class CredentialProvider(Protocol):
     def authorization_header(self) -> tuple[str, str]:
         raise NotImplementedError
 
+    @classmethod
+    def from_config(cls, credential_type: str, config: Any) -> CredentialProvider:
```

Comment:
I suggest we do the same here, use `load`, then expect the following:

```py
dct = {"token": {...}}
credz = CredentialProvider.load(dct)
assert type(credz) is Token

dct = {"client_credentials": {...}}
credz = CredentialProvider.load(dct)
assert type(credz) is OAuthClientCredentials
```

if more than one top-level key exists, probably raise

Reply:
I had that originally and was going to bring it up to discuss 👌🏻